code/controllers/subsystem/fluids/fluid_effects.dm

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ */
+SUBSYSTEM_DEF(fluid_effect)
+	name = "Fluid Effect"
+	wait = 1
+	flags = SS_BACKGROUND|SS_POST_FIRE_TIMING
+
+	//
+	/// The amount of time
+	var/effect_wait = 1 SECONDS
+	///
+	var/list/list/obj/effect/particle_effect/fluid/processing
+	///
+	var/tmp/list/obj/effect/particle_effect/fluid/currentrun
+	///
+	var/tmp/num_buckets
+	///
+	var/tmp/processing_bucket_index
+
+/datum/controller/subsystem/fluid_effect/Initialize(start_timeofday)
+	if (effect_wait < wait)
+		if (effect_wait > 0)
+			wait = effect_wait
+		else
+			effect_wait = wait
+
+	num_buckets = round(effect_wait / wait)
+	effect_wait = wait * num_buckets
+
+	processing = list()
+	processing.len = num_buckets
+	for(var/i in 1 to num_buckets)
+		processing[i] = list()
```

Comment:
```suggestion
	processing = list()
	for(var/i in 1 to num_buckets)
		processing += list()
```
How many buckets do you expect to be making here?

Reply:
The most it makes is current 20: for smoke effects. Otherwise it's either 1 for smoke spreading or 2 for foam spreading and foam effects.

Reply:
Also I'm pretty sure that suggestion will add a total of no buckets to the list because of BYOND list append memes.